src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputModelTypeConverter.cs

```diff
@@ -135,58 +135,65 @@ private static InputModelType CreateInputModelTypeInstance(string? id, string? n
             }
             reader.Read();
 
             while (reader.TokenType != JsonTokenType.EndArray)
             {
-                var property = reader.ReadWithConverter<InputModelProperty>(options);
-                /* TODO: in Multipart body model, if the property is of type Bytes, it should be converted to Stream
-                 * In future, we will convert this in emitter when we adopt tcgc.
-                 */
-                if (property != null && isMultipartType)
+                var rawProperty = reader.ReadWithConverter<InputModelProperty>(options);
+                if (rawProperty == null)
                 {
-                    static InputType ConvertPropertyType(InputType propertyType)
-                    {
-                        return propertyType switch
-                        {
-                            InputPrimitiveType { Kind: InputPrimitiveTypeKind.Bytes } => InputPrimitiveType.Stream,
-                            InputListType listType => new InputListType(listType.Name, listType.CrossLanguageDefinitionId, ConvertPropertyType(listType.ValueType))
-                            {
-                                Decorators = listType.Decorators
-                            },
-                            InputDictionaryType dictionaryType => new InputDictionaryType(dictionaryType.Name, dictionaryType.KeyType, ConvertPropertyType(dictionaryType.ValueType))
-                            {
-                                Decorators = dictionaryType.Decorators
-                            },
-                            InputNullableType nullableType => new InputNullableType(ConvertPropertyType(nullableType.Type))
-                            {
-                                Decorators = nullableType.Decorators
-                            },
-                            _ => propertyType
-                        };
-                    }
+                    throw new JsonException($"Property of model '{modelName}' cannot be null.");
+                }
 
-                    property = new InputModelProperty(property.Name,
-                                                    property.SerializedName,
-                                                    property.Description,
-                                                    ConvertPropertyType(property.Type),
-                                                    property.ConstantValue,
-                                                    property.IsRequired,
-                                                    property.IsReadOnly,
-                                                    property.IsDiscriminator,
-                                                    property.FlattenedNames);
+                var flattenedProperties = FlattenPropertyIfNecessary(rawProperty);
+
+                foreach (var property in flattenedProperties)
+                {
+                    properties.Add(isMultipartType ? ConvertMultipartProperty(property) : property);
```

Comment:
MultipartConvertor is only apply on the first-level property, it should not apply on flattened property of a model.